File: Sr-Navalha-Back/src/main/java/com/fabricasoftware/SrNavalha/models/Agendamento.java
package com.fabricasoftware.SrNavalha.models;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import javax.persistence.*;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@Table(name = "agendamento")
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private long id;
    private Date data;
    private String horario;
    private String status;
    @ManyToOne(cascade = CascadeType.PERSIST)
    private Endereco endereco;
    @OneToOne
    private Servico servico;
    @OneToOne
    private UsuarioCliente cliente;

}

File: Sr-Navalha-Front/src/app/models/Agendamento.ts
import { UsuarioCliente } from './usuario-cliente';
import { Servico } from "./servico";
import { Endereco } from './endereco';

class Agendamento {
    id: any;
    data!: Date;
    formaPagamento: any;
    horario: any;
    status: any;
    endereco!: Endereco;
    servico!: any;
    cliente!: any
}

export { Agendamento };